File: Sem-3/Python/linear_search.py
lst = []
num = int(input("Enter size of list : "))
print("Enter elements of list : ")
for n in range(num):
    numbers = int(input("-> "))
    lst.append(numbers)
x = int(input("Enter number to search in list : "))
i = 0
flag = False
while i < len(lst):
    if lst[i] == x:
        flag = True
        break
    i = i + 1
if flag == 1:
    print('{} was found at index {}'.format(x, i))
else:
    print('{} was not found.'.format(x)) 

File: Sem-3/DSA(C)/Linear_Search.c
#include <stdio.h>
#include <conio.h>

void main()
{
    int len;
    printf("Enter the length of Array: ");
    scanf("%d", &len);
    int arr[len];

    printf("Enter the elements of the array: \n");
    for (int i = 0; i < len; i++)
    {
        scanf("%d", &arr[i]);
    }
    int toSearch;
    printf("Enter the element to search: ");
    scanf("%d", &toSearch);

    int flag = -1;
    // 0 -> Not Found;
    // 1 -> Found;
    for (int i = 0; i < len; i++)
    {
        if (arr[i] == toSearch)
        {
            flag = i;
        }
    }
    if (flag != -1)
    {
        printf("FOUND at index: %d", flag);
    }
    else
    {
        printf("Not Found!!");
    }
}